include/drivers/counter.h

```diff
@@ -366,8 +379,11 @@ static inline u32_t z_impl_counter_read(struct device *dev)
  * @retval -EINVAL if alarm settings are invalid.
  * @retval -ETIME  if absolute alarm was set too late.
  */
-static inline int counter_set_channel_alarm(struct device *dev, u8_t chan_id,
-				      const struct counter_alarm_cfg *alarm_cfg)
+__syscall int counter_set_channel_alarm(struct device *dev, u8_t chan_id,
```

Comment:
this call results in callback being called from interrupt context. I'm not very familiar with user space but it might be forbidden to call user mode function from interrupt context. @andrewboie  is that right?

Reply:
The handler implementation already checks if callback == 0 and bails otherwise.